prov/verbs/src/fi_verbs.c

```diff
@@ -938,29 +954,49 @@ static void fi_ibv_msg_ep_qp_init_attr(struct fi_ibv_msg_ep *ep,
 	}
 }
 
+static inline struct fi_info *fi_ibv_find_info(const char *fabric_name,
+		const char *domain_name)
+{
+	struct fi_info *info;
+	int match = 0;
+
+	for (info = verbs_info; info; info = info->next) {
+		if ((domain_name && !strcmp(info->domain_attr->name, domain_name))
+			|| (fabric_name && !strcmp(info->fabric_attr->name, fabric_name))) {
+			match = 1;
+			break;
+		}
+	}
+
+	return match ? info : NULL;
+}
+
 static int fi_ibv_getinfo(uint32_t version, const char *node, const char *service,
 			  uint64_t flags, struct fi_info *hints, struct fi_info **info)
 {
 	struct rdma_cm_id *id;
 	struct rdma_addrinfo *rai;
-	struct fi_info *fi;
+	struct fi_info *fi, *check_info;
 	int ret;
 
 	ret = fi_ibv_init_info(hints);
 	if (ret)
 		return ret;
 
-	/* TODO When multiple devices are present verbs_info would be list
-	 * of info. In that case we have to choose the correct verbs_info to 
-	 * check against if we have a valid ibv_context in id->verbs */
+	assert(verbs_info);
+
 	ret = fi_ibv_create_ep(node, service, flags, hints, &rai, &id);
 	if (ret)
 		return ret;
 
-	assert(!id->verbs || !strcmp(verbs_info->domain_attr->name,
-				     ibv_get_device_name(id->verbs->device)));
+	if (id->verbs) {
+		check_info = fi_ibv_find_info(NULL, ibv_get_device_name(id->verbs->device));
 
-	if (!(fi = fi_dupinfo(verbs_info))) {
+	} else {
+		check_info = verbs_info;
+	}
```

Comment:
> I shall remove the assert and do away with the check. The exit condition
> for fi_ibv_init_info is verbs_info being set. I guess that's sufficient.

I didn't quite follow this comment.  I was asking to add a check that check_info was not NULL after being set.